homework_4/Barbhaiwala/solution.js

```diff
@@ -0,0 +1,73 @@
+var colors = require("colors");
+
+function HotelParser(jsonString) {
+    
+    this.hotelInfoJson = jsonString; 
+    
+    this.parse = function () { 
+        var parsedHotelInfo = parseHotelInfoFromJson();
+        
+        var hotelList = verify(parsedHotelInfo.hotelList);
+        return hotelList;
+    }
+    
+    function verify(hotelList) {
+        var hotelInformation = [];
+        
+        if (!hotelList) {
+            return hotelInformation;
+        }
+        //for (var hotel in hotelList) {
+        //hotelList.foreach(function (hotel) {
+        for(var idx = 0, hotel; idx < hotelList.length; ++idx) {
+            hotel = hotelList[idx];
+            if (! (hotel.hotelId || hotel.name ||
+                    hotel.address || hotel.city || hotel.stateProvinceCode || hotel.countryCode || hotel.postalCode ||
+                    hotel.hotelStarRating || hotel.hotelGuestRating || hotel.shortDescription )) {
+                break;
+            }
+            
+            hotelInformation.push(
+                new HotelInformation(
+                    hotel.hotelId, hotel.name, 
+                    hotel.address, hotel.city, hotel.stateProvinceCode, hotel.countryCode, hotel.postalCode,
+                    hotel.hotelStarRating, hotel.hotelGuestRating, hotel.shortDescription));
+        }
+        return hotelInformation;
+    }
+    
+    function parseHotelInfoFromJson() {
```

Comment:
You don't need to declare this function inside the `HotelParser`, since it won't be accessible outside of the current module scope. It can get a little messy/confusing when you're declaring functions inside of function like this.